### PR TITLE
Merge master to deep children branch

### PR DESCRIPTION
**Pull request by timse**
master merge branch of #4255

### CODE REVIEW

**Comment by drphelps** [comment]
@timse, @ArcEglos where did we end up with this? Please keep this PR alive.

File: lib/optimize/CommonsChunkPlugin.js [hunk 1]
```diff
@@ -76,6 +77,7 @@ You can however specify the name of the async chunk by passing the desired strin
 			minChunks: options.minChunks,
 			selectedChunks: options.chunks,
 			children: options.children,
+			children: options.deepChildren,
```
Comment (timse) [inline lib/optimize/CommonsChunkPlugin.js]: fix duplicate key "children"

**Comment by timse** [comment]
will take a look tonight

**Comment by ArcEglos** [comment]
Sry I didn't find the time to look into it last weekend. I'm definitely going to keep this alive, just need to find time for it. I hope for tonight but tomorrow is probably more realistic.

**Comment by ArcEglos** [comment]
Sorry for the confusion but I only managed to continue the work in #4255. Don't really have any experience juggling stuff on github

**Comment by stowellt** [comment]
I'm curious if this is still being worked on? I've tried using the code from this pull request for lib/optimize/CommonsChunkPlugin.js and it works great. It looks like the webpack-bot is just reporting a syntax error.

**Comment by Kavenon** [comment]
I would really appreciate adding this feature to future release.

**Comment by ljcrapo** [comment]
I have fixed the issues in yet another pull request: https://github.com/webpack/webpack/pull/5764

**Pull request closed without merging**